stream_alert/rule_processor/threat_intel.py

```diff
@@ -164,6 +166,10 @@ def _extract_ioc_from_record(self, record):
                         for original_key in original_keys:
                             value = value[original_key]
                     if value:
+                        # Threat Intel will only check against public IP addresses
+                        # when process IP IOCs.
+                        if ioc_type == 'ip' and not self.is_public_ip(value):
```

Comment:
why negation logic when you can just do `.is_private()`?

Reply:
see https://viewsby.wordpress.com/2015/02/12/python-check-if-ip-is-public-or-private/

Reply:
We also need to filter out non unicast ip address (multicast, broadcast).

Reply:
(netaddr doc)[https://netaddr.readthedocs.io/en/latest/tutorial_01.html#public] shows how  to check is the ip address is public or not.